### ISSUE TITLE
Parsing a website that has its content inside a cleaned element

### ISSUE THREAD

**Issue by c-st** [postlight/mercury-parser#354]
I'm having an issue with getting this website parsed: https://www.mckinsey.com/industries/financial-services/our-insights/blockchains-occam-problem

After running mercury everything that is getting extracted is the cookie warning. I think the problem is that they decided to place the article content inside a `<form>` element...

For example the xpath expression to get to the headline is `//*[@id="form1"]/div[3]/main/section[1]/div[3]/div/div[1]/h1`. The actual article content is similarly nested inside this form.

It seems that this `<form>` element gets filtered out during a cleanup step. This filtering apparently happens here: https://github.com/postlight/mercury-parser/blob/262dda94b30cf94c180972cbcd7671758c65c9a4/src/resource/utils/dom/constants.js#L6

I started generating a custom extractor but I'm stuck because the actual content is not included in the downloaded html in `fixtures/...` either. 

It seems in the extractor config it's already too late because once the extraction happens the markup is already cleaned up. 

Is there a way to tell mercury to not clean up specific tags from a given extractor? Or do you have any ideas on how to approach this?

**Comment by adampash** [comment]
I think your best option might be to explore transforming that `form` element to something else — namely a `div` — which I believe will work. Let me know if that still doesn't do the job for you.

**Comment by c-st** [comment]
I tried that in the extractor with the following config:

```js
export const WwwMckinseyComExtractor = {
  domain: 'www.mckinsey.com',
  title: {
    selectors: [
      'h1.headline'
    ],
    transforms: {
      form: 'div',
    },
  }
```

It fails however and doesn't find the `h1` (which is located inside the `form`). It looks as if the `form` was already filtered out when extraction starts. 

If I uncommented the line responsible for removing the elements in `src/resource/utils/dom/clean.js` , it works:

```js
export default function clean($) {
  // $(TAGS_TO_REMOVE).remove();
  $ = cleanComments($);
  return $;
}
```

If there was a way for mercury to deal with html (without making the request itself) I could pass it an already transformed response stream. But I don't see any easy way of doing that...
